src/Compilers/Core/Portable/DiagnosticAnalyzer/Suppression.cs

```diff
@@ -48,5 +49,35 @@ public static Suppression Create(SuppressionDescriptor descriptor, Diagnostic su
         /// Diagnostic suppressed by this suppression.
         /// </summary>
         public Diagnostic SuppressedDiagnostic { get; }
+
+        public static bool operator ==(Suppression left, Suppression right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(Suppression left, Suppression right)
+        {
+            return !(left == right);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is Suppression suppression
+                && Equals(suppression);
+        }
+
+        public bool Equals(Suppression other)
+        {
+            return EqualityComparer<SuppressionDescriptor>.Default.Equals(Descriptor, other.Descriptor)
+                && EqualityComparer<Diagnostic>.Default.Equals(SuppressedDiagnostic, other.SuppressedDiagnostic);
+        }
+
+        public override int GetHashCode()
+        {
+            int hashCode = 1755072348;
+            hashCode = hashCode * -1521134295 + EqualityComparer<SuppressionDescriptor>.Default.GetHashCode(Descriptor);
```

Comment:
We have hashcode helpers for combining hashcodes.

Reply:
Switched to using that.